tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,365 @@
+        assert os.path.exists(os.path.join(model_dir, "data_pipeline.py"))
+
+        # 4. Raising ValueError with wrong train_type
+        workspace_path = os.path.join(tmp_dir_path, "otx_cli_builder_wrong")
+        train_type = "unexpected"
+        inputs = {
+            "task_type": "classification",
+            "train_type": train_type,
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        with pytest.raises(ValueError):
+            otx_builder.build_task_config(**inputs)
+
+        # 5. Build workspace with model_type argments
+        workspace_path = os.path.join(tmp_dir_path, "otx_cli_builder_yolox")
+        model_type = "yolox"
+        inputs = {
+            "task_type": "detection",
+            "model_type": model_type,
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        otx_builder.build_task_config(**inputs)
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        template = MPAConfig.fromfile(os.path.join(workspace_path, "template.yaml"))
+        assert template.name.lower() == model_type
+
+        # 6. Raise ValueError when build workspace with wrong model_type argments
+        workspace_path = os.path.join(tmp_dir_path, "otx_cli_builder_atss_2")
+        inputs = {
+            "task_type": "detection",
+            "model_type": "unexpected",
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        with pytest.raises(ValueError):
+            otx_builder.build_task_config(**inputs)
+
+    @e2e_pytest_unit
+    def test_builder_build_backbone_config(self, tmp_dir_path):
+        """
+        <b>Description:</b>
+        Check "Builder.build_backbone_config" function that generate backbone configuration file is working well
+
+        <b>Steps</b>
+        1. Generate backbone config file (mmcls.MMOVBackbone)
+        2. Raise ValueError with wrong output_path
+        """
+        otx_builder = Builder()
+
+        # 1. Generate backbone config file (mmcls.MMOVBackbone)
+        tmp_backbone_path = os.path.join(tmp_dir_path, "backbone.yaml")
+        backbone_type = "mmcls.MMOVBackbone"
+        otx_builder.build_backbone_config(backbone_type, tmp_backbone_path)
+        assert os.path.exists(tmp_backbone_path)
+        backbone_config = mmcv.load(tmp_backbone_path)
+        assert backbone_config["backbone"]["type"] == backbone_type
+
+        # 2. Raise ValueError with wrong output_path
+        tmp_backbone_path = os.path.join(tmp_dir_path, "wrong.path")
+        with pytest.raises(ValueError):
+            otx_builder.build_backbone_config(backbone_type, tmp_backbone_path)
+
+    @e2e_pytest_unit
+    def test_builder_build_model_config(self, tmp_dir_path):
+        """
+        <b>Description:</b>
+        Check "Builder.build_model_config" function that update model config with new backbone is working well
```

Comment:
Probably does not relate to the unit test itself, but the design.

so this function:`Builder.build_model_config(self, model_config_path, backbone_config_path)` builds a model from the model config path. And is it true that it also updates the backbone of the model? 

Wouldn't it makes more sense if we build a model from OTX model builder and call `otx_builder.udpate_backbone(backbone_type: str)` to update the backbone in the model instance?

Reply:
Yes, the inputs (model.py & backbone.yaml) and outputs (model.py) of this are configurations, but actually build backbone, measure the channels using fake data, and update the necessary config value of the model.
we can think of it as a function that `merges the model and the backbone` and `emits the updated model`.

It's probably okay to configure it with a function named `update_backbone`, but the concern is that we aim to update based on the `backbone's configuration file`, not targeting the `backbone_type`. so it doesn't seem appropriate to receive only `backbone_type` as an argment. Or it seems that it can be divided into separate functions inside.

To sum up, there are two ideas for a more understandable design: 
1. Rename this function to `update_backbone` (same inside code)
2. How to build a file config and put a separate function inside to reduce redundant code and split it into two functions. (`build_model_config` -> call `update_backbone(backbone_type: str)`) - I'll have to check if this is possible or good design while implementing it myself.

What is your recommendation?

Reply:
> What is your recommendation?

I think it'd be better if we call it `merge_backbone`, and keep the parameters as it is now. 

A remark while I was trying builder out:  
It's a great feature! That you can configure a model in this way. However, in terms of user experience, I feel model building is interactive and CLI is limited to the interactive process. I've been looking into [pytermgui](https://github.com/bczsalba/pytermgui) and [textual](https://github.com/Textualize/textual), they are terminal GUI. I hope we can bring this to another level i.e. building a model in terminal GUI. Maybe not now, but great if we add this feature.

Reply:
Your opinion looks good. I renamed `build_model_config` to `merge_backbone`.
I also think it is difficult to maximize and show our strengths with the current CLI.
It would be nicer if the combination of OTX workspace and several commands would work as a terminal GUI like you suggested. (Guess there's a lot more to think about!)

thanks for the comment :)